src/dispatch/database/revisions/tenant/versions/2024-06-28_15a8d3228123.py

```diff
@@ -0,0 +1,128 @@
+        sa.Column("editable", sa.Boolean(), nullable=True),
+        sa.Column("search_vector", sqlalchemy_utils.types.ts_vector.TSVectorType(), nullable=True),
+        sa.Column("project_id", sa.Integer(), nullable=True),
+        sa.Column("updated_at", sa.DateTime(), nullable=True),
+        sa.Column("created_at", sa.DateTime(), nullable=True),
+        sa.ForeignKeyConstraint(["project_id"], ["project.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index(
+        "case_cost_type_search_vector_idx",
+        "case_cost_type",
+        ["search_vector"],
+        unique=False,
+        postgresql_using="gin",
+    )
+    op.create_table(
+        "case_cost",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("amount", sa.Numeric(precision=10, scale=2), nullable=True),
+        sa.Column("case_cost_type_id", sa.Integer(), nullable=True),
+        sa.Column("case_id", sa.Integer(), nullable=True),
+        sa.Column("project_id", sa.Integer(), nullable=True),
+        sa.Column("updated_at", sa.DateTime(), nullable=True),
+        sa.Column("created_at", sa.DateTime(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["case_cost_type_id"],
+            ["case_cost_type.id"],
+        ),
+        sa.ForeignKeyConstraint(["case_id"], ["case.id"], ondelete="CASCADE"),
+        sa.ForeignKeyConstraint(["project_id"], ["project.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_table(
+        "assoc_signal_instance_tags",
+        sa.Column("signal_instance_id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.Column("tag_id", sa.Integer(), nullable=False),
+        sa.ForeignKeyConstraint(["signal_instance_id"], ["signal_instance.id"], ondelete="CASCADE"),
+        sa.ForeignKeyConstraint(["tag_id"], ["tag.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("signal_instance_id", "tag_id"),
+    )
+    op.add_column("case_type", sa.Column("cost_model_id", sa.Integer(), nullable=True))
+    op.create_foreign_key(None, "case_type", "cost_model", ["cost_model_id"], ["id"])
+    op.alter_column(
+        "entity", "source", existing_type=sa.BOOLEAN(), type_=sa.String(), existing_nullable=True
+    )
+    op.drop_index("ix_entity_search_vector", table_name="entity", postgresql_using="gin")
+    op.create_index(
+        "entity_search_vector_idx",
+        "entity",
+        ["search_vector"],
+        unique=False,
+        postgresql_using="gin",
+    )
+    op.alter_column("entity_type", "jpath", existing_type=sa.VARCHAR(), nullable=True)
+    op.add_column("participant_activity", sa.Column("case_id", sa.Integer(), nullable=True))
+    op.create_foreign_key(None, "participant_activity", "case", ["case_id"], ["id"])
+    op.drop_column("plugin_instance", "configuration")
+    op.drop_constraint("project_stable_priority_id_fkey", "project", type_="foreignkey")
```

Comment:
Are all these schema changes correct / related to the current changes? Some that look out of place are:

```python
    op.alter_column(
        "entity", "source", existing_type=sa.BOOLEAN(), type_=sa.String(), existing_nullable=True
    )
    op.drop_index("ix_entity_search_vector", table_name="entity", postgresql_using="gin")
    op.create_index(
        "entity_search_vector_idx",
        "entity",
        ["search_vector"],
        unique=False,
        postgresql_using="gin",
    )
    op.alter_column("entity_type", "jpath", existing_type=sa.VARCHAR(), nullable=True)
```

```python
    op.drop_column("plugin_instance", "configuration")
    op.drop_constraint("project_stable_priority_id_fkey", "project", type_="foreignkey")
```